stellar/ttl.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const { Contract, SorobanRpc, SorobanDataBuilder, TransactionBuilder, Operation } = require('@stellar/stellar-sdk');
+const { Server } = require('@stellar/stellar-sdk/rpc');
+const { Command } = require('commander');
+const { execSync } = require('child_process');
+const { loadConfig, printInfo, saveConfig } = require('../evm/utils');
+const { stellarCmd, getNetworkPassphrase, getWallet, addBaseOptions } = require('./utils');
+const { getChainConfig, addOptionsToCommands } = require('../common');
+const { prompt } = require('../common/utils');
+require('./cli-utils');
+
+const MAX_INSTANCE_TTL_EXTENSION = 535679;
+
+function checkContractName(chain, contractName) {
+    if (!chain.contracts?.[contractName]) {
+        throw new Error('Contract not found.');
+    }
+}
+
+async function getTtl(chain, contractName, _args, _options) {
+    printInfo('get ttl', contractName);
+    const ledgerEntry = await getLedgerEntry(chain, contractName);
+    printInfo('latest ledger', ledgerEntry.latestLedger);
+    printInfo('live until ledger', ledgerEntry.entries[0].liveUntilLedgerSeq);
+}
+
+async function getLedgerEntry(chain, contractName) {
+    const contract = new Contract(chain.contracts[contractName].address);
+    const instance = contract.getFootprint();
+    const server = new SorobanRpc.Server(chain.rpc);
+    return await server.getLedgerEntries(...[instance]);
+}
+
+async function extendInstance(chain, contractName, _args, options) {
+    checkContractName(chain, contractName);
+
+    const { yes } = options;
+    const { rpc, networkType } = chain;
+
+    if (prompt(`Extend instance ttl for ${contractName}`, yes)) {
+        return;
+    }
+
+    const server = new Server(rpc);
+    const wallet = await getWallet(chain, options);
+
+    const account = await server.getAccount(wallet.publicKey());
+    const fee = '200100'; // Base fee plus resource fee
+
+    const contract = new Contract(chain.contracts[contractName].address);
+    const instance = contract.getFootprint();
+
+    const sorobanData = new SorobanDataBuilder().setResourceFee(200000).setReadOnly([instance]).build();
+    const transaction = new TransactionBuilder(account, {
+        fee,
+        networkPassphrase: getNetworkPassphrase(networkType),
+    })
+        .setSorobanData(sorobanData)
+        .addOperation(
+            Operation.extendFootprintTtl({
+                extendTo: MAX_INSTANCE_TTL_EXTENSION,
+            }),
+        )
+        .setTimeout(30)
+        .build();
+
+    transaction.sign(wallet);
+    await server.sendTransaction(transaction);
+}
+
+async function restoreInstance(chain, contractName, _args, options) {
+    checkContractName(chain, contractName);
```

Comment:
I believe this is being checked in `mainProcessor` function, no need to double check.